build.js

```diff
@@ -1611,10 +1612,24 @@ var buildExecutable = function(rootPackageName, buildPackagesResultsCache, resou
   var packageResource = resourceCache[rootPackageName];
   var executableArtifact = buildForExecutable(packageResource, packageResource, buildConfig);
   var findlibLinkCommand = getFindlibCommand(packageResource, programForCompiler(buildConfig.compiler), true);
-  var justTheModuleArtifactsForAllPackages =
+  var allModuleArtifacts =
     getJustTheModuleArtifactsForAllPackages(rootPackageName, buildPackagesResultsCache, resourceCache);
 
 
+  var usersModuleArtifacts = allModuleArtifacts
```

Comment:
This all seems kind of hacky. Do you think it would be cleaner just to write a better version of `getJustTheModuleArtifactsForAllPackages` that returns just what you want in the form you want it?